usaspending_api/common/operations_reporter.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+
+class OpsReporter:
+    required_files = ("job_name", "duration", "iso_start_datetime", "end_status")
+
+    def __init__(self, **kwargs):
+        self.internal_dict = {}
+        if kwargs:
+            self.internal_dict = kwargs
+
+    def __getattr__(self, key):
+        try:
+            return self.internal_dict[key]
+        except KeyError:
+            raise AttributeError(key)
+
+    def __getitem__(self, key):
+        try:
+            return self.internal_dict[key]
+        except KeyError:
+            raise AttributeError(key)
+
+    def __setitem__(self, key, value):
+        self.internal_dict[key] = value
+
+    def __str__(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return ",".join(["{}:{}".format(k, v) for k, v in self.internal_dict.items()])
+
+    def json_dump(self):
+        self._verify_required_fields()
+        return json.dumps(self.internal_dict)
+
+    def _verify_required_fields(self):
+        for field in self.required_files:
+            self.internal_dict[field]
```

Comment:
This is slightly confusing to me... Is this intended to just raise a KeyError if internal_dict does not contain the field?

Reply:
Yes

Reply:
My only comment, and its not essential, but maybe json_dump should have a try/except that raises something else that explicitly communicates that the internal dict is missing a required field.  But I'll approve, as this is probably just nitpicking.